src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -134,7 +134,16 @@ protected boolean submit(O op) throws InterruptRunException, IllegalStateExcepti
                         // if(success) return true;
                         break;
                     case READ:
-                        // if(success) return true;
+                    	final String path = pathOp.dstPath();
+						final String scopeName = path.substring(0,path.indexOf("/"));
+						final String streamName = path.substring(path.indexOf("/")+1, path.length());
+						if (scopeMap.get(scopeName).get(streamName)==null) {
+							//putIf or computeIf
+							//probably put inside to avoid code repeat
+						};
+                        if(invokePathRead((PathOperation<? extends PathItem>) op)) {
```

Comment:
Changed. Agree. But to avoid an unchecked assignment then we need to cast like this:
`final PathOperation<? extends PathItem> pathOp = (PathOperation<? extends PathItem>) op;`
in else branch for PathItem.